tests/scale/virt-api.go

```diff
@@ -45,12 +73,28 @@ var _ = Describe("[sig-compute] virt-api scaling", decorators.SigCompute, func()
 		return expectedReplicas
 	}
 
-	It("virt-api replicas should be scaled as expected", func() {
+	DescribeTable("virt-api replicas should be scaled as expected", func(withJsonPatch bool) {
 		By("Finding out nodes count")
 		nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), v1.ListOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
 		numberOfNodes := len(nodes.Items)
+		expectedResult := calcExpectedReplicas(numberOfNodes)
+		if withJsonPatch {
+			expectedResult += 1
+			ccs := v12.CustomizeComponents{
+				Patches: []v12.CustomizeComponentsPatch{
+					{
+						ResourceName: components.VirtAPIName,
+						ResourceType: "Deployment",
+						Patch:        `[{"op":"replace","path":"/spec/replicas","value":` + strconv.Itoa(int(expectedResult)) + `}]`,
+						Type:         v12.JSONPatchType,
+					},
+				},
+			}
+			oldcss := setccs(ccs)
+			defer restorescc(oldcss)
```

Comment:
Why is it better can you please elaborate?

Reply:
`defer` will run at the end of the scope of the function, while `DeferCleanup` will create, summarizing, an `AfterEach` block (https://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup).
So, `DeferCleanup` is created ad hoc for these cases:  I am not sure, but `defer` also may conflict with the report function since the deferred function is out of scope of the ginkgo suite.

Reply:
I couldn't find any advantage in the current code.

Reply:
I still suggest you to use `DeferCleanup` or `AfterEach` https://www.kubernetes.dev/blog/2023/04/12/e2e-testing-best-practices-reloaded/#recovering-from-test-failures